pam/internal/adapter/pam.proto

```diff
@@ -0,0 +1,18 @@
+syntax = "proto3";
+package pam;
+
+option go_package = "github.com/ubuntu/authd/pam/internal/adapter";
+
+// state represents the stage object for the UI.
+enum Stage {
+    // Stage_undefined represents an undefined stage.
+    undefined = 0;
```

Comment:
The idea in Go is that the zero value is useful. I think then the 0 value here should be for the first actionnable stage: userselection.

Do you use undefined explicitely for anything?

Reply:
Yeah, in order to check whether the stage has been changed first time, otherwise the first time we call `changeStage` it would be set to `userSelection`, and that's wrong, since the first state is unset. Not having this would break the logic to detect if the state has changed.

Now, I'd have avoided to set it completely, and just start the enum from 1, but: `pam.proto:9:21: The first enum value must be zero in proto3.`.

So, we need to keep it, not sure if naming could be different, but in general I feel that a variable by default should be in an undefined state, then once assigned has a value.

Reply:
> So, we need to keep it, not sure if naming could be different, but in general I feel that a variable by default should be in an undefined state, then once assigned has a value.

Again, this is not idiomatic, the zero value should be useful in Go:
- https://go-proverbs.github.io/
- https://www.youtube.com/watch?v=PAAkCSZUG1c&t=385s (from the creator of Go)
- https://jerryan.medium.com/make-your-zero-value-useful-in-golang-2771ada86b53
- https://dave.cheney.net/2013/01/19/what-is-the-zero-value-and-why-is-it-useful
- https://victorpierre.dev/articles/zero-values-in-go/

So, let’s try to not work against the language paradigm. I thin the code can assume that we are in user selection by default, and not in an undefined state, and then, if we autoselect, we can move to another stage and change it.

Reply:
> Again, this is not idiomatic, the zero value should be useful in Go:

I see this, but that's a proto file that wasn't supposed to be used only in go, thus the reason... However, I will change it, but it's causing problem to the new `changeState` logic, so it should be re-adjusted.

Reply:
At the init of the program, we are in the user selection stage and should prepare for this. Then, we don’t need to cast an unknown stage.

Reply:
and on the proto, this is why I didn’t want first to have an enum, you insisted on it, but at least, let’s keep the code idiomatic to the main language in use for the project :)

Reply:
> At the init of the program, we are in the user selection stage and should prepare for this. Then, we don’t need to cast an unknown stage.

We are at the start, the problem is when going back to the state IIRC, in fact `m.userSelectionModel.Focused()` may not be true and that's of the issues, but I need to check better the details, for sure the fact we may return we're in a state that is not really the one we're in from an UI POV, thus the issues.

Reply:
Ok, the errors I got were due to the inconsistent state of my system, that was slowing down the tests, and thus making them fail (damned `sleep`'s! :)).

Reply:
So... I went back to using `0`.

Reply:
I don’t think this is a valid use case :) I suggest that the system (as it used to be in the CLI side), to be in a working state by default in this selection user mode, and not delivering a signal to change state from the state.

Basically initial state == working state -> no signal by default.